bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java

```diff
@@ -186,6 +188,62 @@ private String getTransactionLogDirectory() {
         }
     }
 
+    /**
+     * This method gets the user specified config for the transaction auto commit timeout. Default is 120.
+     *
+     * @return int transaction auto commit timeout value
+     */
+    public int getTransactionAutoCommitTimeout() {
+        VariableKey transactionAutoCommitTimeoutKey = new VariableKey(TRANSACTION_PACKAGE_ID,
+                "transactionAutoCommitTimeout", PredefinedTypes.TYPE_INT, false);
+        if (!ConfigMap.containsKey(transactionAutoCommitTimeoutKey)) {
+            return DEFAULT_TRX_AUTO_COMMIT_TIMEOUT;
+        } else {
+            int transactionAutoCommitTimeout;
+            Object value = ConfigMap.get(transactionAutoCommitTimeoutKey);
+            if (value instanceof Long) {
+                transactionAutoCommitTimeout = ((Long) value).intValue();
+            } else if (value instanceof Integer) {
+                transactionAutoCommitTimeout = (Integer) value;
+            } else {
+                return DEFAULT_TRX_AUTO_COMMIT_TIMEOUT;
+            }
+            if (transactionAutoCommitTimeout < 0) {
+                return DEFAULT_TRX_AUTO_COMMIT_TIMEOUT;
+            } else {
+                return transactionAutoCommitTimeout;
+            }
```

Comment:
Since we duplicate this logic in both `getTransactionAutoCommitTimeout` and `getTransactionCleanupTimeout`, we can separate it into a different function.